### DISCUSSION TITLE
The index is not updated in the loop

### DISCUSSION THREAD

**Discussion by Dzemy** [livewire/livewire#7527] [1. Help]
Can I please ask how the $index variable inside the "item" component is not updated if I change their order?

```php
public array = [1,2,3,4,5,6,7]
```

```blade
<div>
    @foreach($array as $index => $item)
        <livewire:item wire:key="{{$item}}" :$item :$index />
    @endforeach
</div>
```

The item variable is updated to the new one.

**Comment by joshhanley** [comment]
@Dzemy the issue will probably be the `wire:key`, you need to make sure they key is unique (and not the index of the loop) to ensure Livewire can track it property. So I would use the item ID or something like that.

Hope this helps!

**Reply by Dzemy** [reply]
I got it wrong, wire:key should be $item, I got it right in the code. My array is [1,2,3,4,5,6,7] so I used that number as the key.
